libsolidity/codegen/ExpressionCompiler.cpp

```diff
@@ -1317,10 +1359,126 @@ void ExpressionCompiler::appendArithmeticOperatorCode(Token::Value _operator, Ty
 		m_context << Instruction::SUB;
 		break;
 	case Token::Mul:
-		m_context << Instruction::MUL;
+		if (c_isFractional) //fixed point handling
+		{
+			if (c_numBits - c_fractionalBits == 0)
+			{
+				//take two numbers, cut them in half, multiply them, simple as that.
+				m_context << c_halfShift << Instruction::DUP1 << Instruction::SWAP2 << (c_isSigned ? Instruction::SDIV : Instruction::DIV); 
+				m_context << Instruction::SWAP2 << (c_isSigned ? Instruction::SDIV : Instruction::DIV) << Instruction::MUL;
+
+			}
+			else if (c_numBits > 128)
+			{
+				//time to get schwifty in here...
+				//split both numbers into their representative fractional and decimal parts
+				//this takes the form A.B * C.D
+				//then we utilize this formula:
+				//D*B/shift + C*A*shift + D*A + C*B 
+				//m_context << Instruction::DUP1 << Instruction::DUP3;
+				//D*B/shift
+				//m_context << c_fractionShift << Instruction::SWAP1 << (c_isSigned ? Instruction::SMOD : Instruction::MOD);
+				//m_context << Instruction::SWAP1 << c_fractionShift <<  Instruction::SWAP1 << (c_isSigned ? Instruction::SMOD : Instruction::MOD);
+				if (c_fractionalBits > c_intBits)
+				{
+					//we need this here because overwise we will have an overflow with our fractional bits
+
+					appendInlineAssembly(R"(
+						{
+							let firstNum := dup1
```

Comment:
If this does not work, then I think we need to change the inline assembly parser